core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/heuristic/move/CompositeMove.java

```diff
@@ -102,6 +102,11 @@ public CompositeMove<Solution_> doMove(ScoreDirector<Solution_> scoreDirector) {
         return new CompositeMove<>(undoMoves);
     }
 
+    @Override
+    public void doMoveOnly(ScoreDirector<Solution_> scoreDirector) {
+        Arrays.stream(moves).filter(move -> move.isMoveDoable(scoreDirector)).forEach(move -> move.doMoveOnly(scoreDirector));
```

Comment:
I recommend not using the streams here, rather switch to a normal for loop.
`doMoveOnly(...)` is on the solver hot path, and therefore we need the fastest possible way of iterating.
(Streams have a lot of overhead. Good for one-time work, bad on the hot path.)